src/arrayMethodJoin.js

```diff
@@ -4,8 +4,29 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedArr = '';
+    let pseudoSeparator = separator;
+
+    if (separator === null) {
+      pseudoSeparator = 'null';
+    }
+
+    for (const n of this) {
+      let item = n + '';
+
+      if (typeof n === 'undefined' || n === null) {
```

Comment:
it'll never be equal to `null` after the conversion to string, so that's why you should remove this `let item = n +''` code

Reply:
But there is a test, saying that it shouldn't add new items if they are null (not 'null'). How to check on null without `n === null`?
<img width="552" alt="image" src="https://user-images.githubusercontent.com/122180564/219874632-3acede75-a124-430b-99d3-e63dde75232d.png">

Reply:
Just filter the array after?🤔